src/iterator.cr

```diff
@@ -297,30 +297,26 @@ module Iterator(T)
   # * If *reuse* is given, the array can be reused
   # * If *reuse* is an `Array`, this array will be reused
   # * If *reuse* is truthy, the method will create a new array and reuse it.
+  # * If *reuse* is falsey, the array will not be reused.
   #
   # This can be used to prevent many memory allocations when each slice of
   # interest is to be used in a read-only fashion.
   def cons(n : Int, reuse = false)
     raise ArgumentError.new "Invalid cons size: #{n}" if n <= 0
-    Cons(typeof(self), T, typeof(n)).new(self, n, reuse)
+    if reuse.is_a?(Bool)
```

Comment:
If reuse is nil this might be falling in the undesired branch. The docs state about falsey values.

Reply:
Good point. Added a check for nil and a test for that case.